### DISCUSSION TITLE
Does trilium handle one note with a lot of large-ish images well?

### DISCUSSION THREAD

**Discussion by wadbr** [TriliumNext/Trilium#1580] [Q&A]
Hello! I love the app and use it often so thank you for everything! 

I wanted to ask first before i fry my db, will one note filled with thousands~ of 1080x1920 screenshots (in png, high quality)? 

I'm fine with making them into multiple notes but would prefer if it was fine in one note.

**Comment by zadam** [comment]
Hi, I haven't tried but my guess is no, it won't handle it well.

Limiting factors are that CKEditor (the editing component) does not handle very long notes well. If it's a read only note then it's going to be better but with really thousands of such images the blink/chromium will struggle.

**Reply by wadbr** [reply]
Gotcha! Thanks zadam! 

That is what I thought, well I will try it with a test db with a couple thousand images in a single note for science and report back my results in a couple of days. 

Thanks again! It takes a lot to make a unique and beautiful note app in an ocean of note taking crud apps <3

**Comment by wadbr** [comment]
Sorry about the late reply, had some personal stuff to attend to. So after testing this for a while it seems Trilium handles a moderate amount of files very well (it lags when first opening the notes but if you are working on a seperate note it's perfectly fine). 

Although I reached around 5k photos perfectly fine, the photos were screenshots that aren't full resolution/big in size (around 400kb max and 200k average) so your mileage may vary! 

@zadam quick question, I noticed when exporting the notes with the large amount of pictures that trilium compresses the photos extremely, by 90% in some cases. Is there any way to stop this behavior and have no compressing of the photos? Or is this a default ck editor behavior?

**Reply by zadam** [reply]
Images are compressed during import. You can tweak the settings here:

![image](https://user-images.githubusercontent.com/617641/107685584-73e8b900-6ca4-11eb-98a3-4783dfdfd0f4.png)

This can be also disabled during import by unchecking "shrink images":

![image](https://user-images.githubusercontent.com/617641/107685729-9e3a7680-6ca4-11eb-906b-eefb4c244d61.png)

**Reply by wadbr** [reply]
Oh I thought that was for jpg only, because the images i got in the export were png (images during import were also png)
